feature/quicksettings/src/main/java/com/google/jetpackcamera/feature/quicksettings/ui/QuickSettingsComponents.kt

```diff
@@ -0,0 +1,225 @@
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.google.jetpackcamera.feature.quicksettings.CameraAspectRatio
+import com.google.jetpackcamera.feature.quicksettings.CameraFlashMode
+import com.google.jetpackcamera.feature.quicksettings.CameraLensFace
+import com.google.jetpackcamera.quicksettings.R
+import com.google.jetpackcamera.settings.model.FlashModeStatus
+import kotlin.math.min
+
+
+// completed components
+
+@Composable
+fun ExpandedQuickSetRatio(
+    setRatio: (/*TODO: insert ratio function*/) -> Unit,
+    currentRatio: Int
+) {
+    val buttons: Array<@Composable () -> Unit> = arrayOf(
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 1, currentRatio = currentRatio) },
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 2, currentRatio = currentRatio) },
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 3, currentRatio = currentRatio) }
+
+    )
+    ExpandedQuickSetting(quickSettingButtons = buttons)
+}
+
+@Composable
+fun QuickSetRatio(onClick: () -> Unit, ratio: Int, currentRatio: Int) {
+    val enum = when (ratio) {
+        1 -> CameraAspectRatio.THREE_FOUR
+        2 -> CameraAspectRatio.NINE_SIXTEEN
+        3 -> CameraAspectRatio.ONE_ONE
+        else -> CameraAspectRatio.ONE_ONE
+    }
+    QuickSettingUiItem(
+        drawableResId = enum.getDrawableResId(),
+        text = stringResource(id = enum.getTextResId()),
+        accessibilityText = stringResource(id = enum.getDescriptionResId()),
+        onClick = { onClick() },
+    )
+}
+
+@Composable
+fun QuickSetFlash(onClick: (FlashModeStatus) -> Unit, currentFlashMode: FlashModeStatus) {
+    val enum = when (currentFlashMode) {
+        FlashModeStatus.OFF -> CameraFlashMode.OFF
+        FlashModeStatus.AUTO -> CameraFlashMode.AUTO
+        FlashModeStatus.ON -> CameraFlashMode.ON
+    }
+    QuickSettingUiItem(
+        drawableResId = enum.getDrawableResId(),
+        text = stringResource(id = enum.getTextResId()),
+        accessibilityText = stringResource(id = enum.getDescriptionResId()),
+        isHighLighted = currentFlashMode == FlashModeStatus.ON,
+        onClick =
+        {
+            when (currentFlashMode) {
+                FlashModeStatus.OFF -> onClick(FlashModeStatus.ON)
+                FlashModeStatus.ON -> onClick(FlashModeStatus.AUTO)
+                FlashModeStatus.AUTO -> onClick(FlashModeStatus.OFF)
+            }
+        }
+    )
+}
+
+@Composable
+fun QuickFlipCamera(flipCamera: (Boolean) -> Unit, currentFacingFront: Boolean) {
```

Comment:
I think it's better we use enum instead of boolean here. There can potentially be more than two cameras in the future if we take into account external camera or concurrent cameras

Reply:
There could be multiple cameras to select from, but this is is to change which direction we are facing, which could either be front or back